### ISSUE TITLE
Dev can't see problem with screens that use secrets. What to do?

### ISSUE THREAD

**Issue by plocket** [plocket/docassemble-cucumber#677]
This may take some discussion. A user had an error for a sign in page and wasn't sure how to troubleshoot what happened. They had the wrong values for email or password.

For sign in pages specifically, we might be able to improve the error message if we can detect more about the error. For example, that the user name and password didn't work vs. not being able to continue because the server was unresponsive. That would also speed up failures for that kind of page. I'll make that a separate issue.

This is a problem with secret variables in general, though. With the above sign in problem, a screenshot could have helped a lot. Should we create a flag to allow dev to take screenshots of screens with secrets? Maybe something they can run as a manual test? We could tell them to delete immediately after, but it seems unsecure anyway. Is there some other way we can help them?

**Comment by BryceStevenWilley** [comment]
IMO, manual tests are the best solution; GitHub Actions won't censor images, so I'm hesitant about letting people take pictures of secrets.

It would be cool if we don't do censoring on the Playground tests though, since that'll be the easiest way for most folks to run Kiln tests locally.

**Comment by plocket** [comment]
Manual tests don't help if the values they provided to the tests are wrong, and that's what was going on in this case - that's what I'm trying to troubleshoot.

> It would be cool if we don't do censoring on the Playground tests though, since that'll be the easiest way for most folks to run Kiln tests locally.

I like this idea, though I wonder if we should put it under a flag. Otherwise they'll have different behaviors on GitHub vs. locally and I generally try to avoid inconsistency. Also, though, it wouldn't help with the problem that I was describing as the values were in GitHub env vars, so the local tests wouldn't show that.

**Comment by BryceStevenWilley** [comment]
> Manual tests don't help if the values they provided to the tests are wrong, and that's what was going on in this case - that's what I'm trying to troubleshoot.

Poor phrasing on my part; by manual tests, I meant running ALKiln locally, in dev mode, where they can see what's getting entered for the email and the password.

> it wouldn't help with the problem that I was describing as the values were in GitHub env vars, so the local tests wouldn't show that.

It wouldn't help because before now, most folks only option was to run through GitHub. Once KilnInThePlayground is ready, we could encourage people to write their tests in the playground, and when they're passing then, commit and run with GitHub actions. 

I'm just really hesitant to add flags that could potentially expose secrets to the public, and it's pretty easy to do stuff like that with Actions.

**Comment by plocket** [comment]
> I'm just really hesitant to add flags that could potentially expose secrets to the public, and it's pretty easy to do stuff like that with Actions.

The flag would just be about Playground behavior. That is `if source is the playground and some flag is true: take screenshot of secrets`. [At least, that was my second suggestion. I agree that the first suggestion - just one flag for everywhere - was making me nervous too.]

**Comment by plocket** [comment]
So I just took a screenshot of a secret with this:
```
Feature: chair

Scenario: frog
  Given I start the interview at "env_var_practice.yml"
  And I set the var "env_var" to the GitHub secret "EXAMPLE"
  Then I take a screenshot
```

I have to check up on how we set things up and see if this is something we explicitly prevented. I don't think so. If not, do we need to prevent it? That is, by preventing all screenshots if as soon as _any_ secret variables are used _anywhere_ in a Scenario? Otherwise, we should document that people should be super careful about this.
